dcpy/connectors/edm/recipes.py

```diff
@@ -47,7 +53,9 @@ def s3_folder(self) -> str:
     def s3_key(self) -> str:
         return f"{self.s3_folder}/{self.file_name}"
 
-    def assign_file_type(self, file_type_preferences: list[DatasetType]) -> DatasetType:
+    def assign_file_type(
```

Comment:
This object's mutability feels a little bit awkward... I wonder if makes more sense for this dataset object to simply list the available file types that it finds on s3, instead of specifying one. (so we'd change `file_type` to be `file_types`, or a method called `get_file_types`). Then we can leave the assigning for the `plan.plan_recipe` function

Reply:
That makes sense, I'll try something out